src/components/Contact.tsx

```diff
@@ -18,7 +18,53 @@
 import React from "react";
 import { RiPhoneFill, RiMailFill } from "react-icons/ri";
 
+import {useState} from 'react'
+import axios from 'axios';
+
+import "react-toastify/dist/ReactToastify.css"
+import {toast, ToastContainer} from "react-toastify"
+
 export default function Example() {
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [message, setMessage] = useState('')
+  const [phone, setPhone] = useState('')
+
+  const [disableButtonState, setDisableButtonState] = useState(false)
+
+  async function onSubmit(event: any) {
+    event.preventDefault()
+
+    const formToBeSubmitted = {
+      name: name,
+      email: email,
+      message: message + "\n\nPhone: " + phone
+    }
+
+    setDisableButtonState(true)
+
+    try {
+      toast.promise(
+        axios.post("https://email-free-api.herokuapp.com/send-email", formToBeSubmitted),
+        {
+          pending: {
+            render(){return 'Sending email...'}
+          },
+          success: {
+            render(){return '🐈‍⬛ Email Sent!'}
+          },
+          error: {
+            render(){return '🐈‍⬛ Error!'}
+          }
+        }
+      )
+    } catch (error) {
+      //if(error) errorToastEmitter()
+    }
```

Comment:
I think try catch it's not necessary. But I don't have 100% of certainty.

Reply:
I just tested it here, seems like it is necessary to catch the error. The button keeps disabled if an error is returned withou the try/catch block.

Reply:
okay